handlers/public.go

```diff
@@ -0,0 +1,169 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/payments.api.ch.gov.uk/config"
+	"github.com/companieshouse/payments.api.ch.gov.uk/dao"
+	"github.com/companieshouse/payments.api.ch.gov.uk/models"
+	"github.com/gorilla/pat"
+)
+
+type paymentService struct {
```

Comment:
Shouldn't this be in a separate file to separate the layers? handler (HTTP handling) -> service (business logic) -> DAO (storage)